x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/cytoscape.tsx

```diff
@@ -98,11 +108,24 @@ export function Cytoscape({ children, elements, height, style, width }: Cytoscap
   // Trigger a custom "data" event when data changes
   useEffect(() => {
     if (cy) {
-      cy.add(elements);
+      if (itemsDeleted === false) {
+        cy.add(elements);
+      } else if (itemsDeleted === true) {
+        cy.elements().remove();
+        cy.add(elements);
+      }
+
       cy.trigger('data');
     }
   }, [cy, elements]);
 
+  // Reset the graph to original zoom and pan
+  useEffect(() => {
+    if (cy) {
+      cy.reset();
+    }
+  }, [cy, resetCy]);
```

Comment:
this flow with `resetCy` doesn't seem straightforward to me. Have you considered creating a service for managing `cy` and putting it into react context?

Reply:
Yeah, I think that sounds like a good idea for a refactor - not in the scope of this PR, though.